modules/project/variables.tf

```diff
@@ -302,10 +302,12 @@ variable "shared_vpc_service_config" {
   description = "Configures this project as a Shared VPC service project (mutually exclusive with shared_vpc_host_config)."
   # the list of valid service identities is in service-agents.yaml
   type = object({
-    host_project                = string
-    service_identity_iam        = optional(map(list(string)), {})
-    service_identity_subnet_iam = optional(map(list(string)), {})
-    service_iam_grants          = optional(list(string), [])
+    host_project                 = string
+    host_project_iam             = optional(list(string), [])
```

Comment:
Using a list for type will make it very fragile as dynamic values in keys usually trigger an error on first apply, which was one of the reasons for not supporting this . :) We should mention it in the README (which you might already do, haven't gotten that far yet).

Reply:
Added warning for  that attribute! Thanks